mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go

```diff
@@ -30,6 +31,13 @@ func dataSourceMongoDBAtlasCloudProviderAccessSetup() *schema.Resource {
 			"aws": {
 				Type:     schema.TypeMap,
 				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"aws_config": {
+				Type:     schema.TypeList,
```

Comment:
[Question] For my own understanding and learning. Why use `TypeList with maxlen 1` instead of TypeStr?

Reply:
Because TypeList with MaxItems 1 ideally it's a object from the API, and TypeSet is an unordered list

Reply:
ah, I miss typed it. I meant why not use `TypeString`?

Reply:
:O, I misread it then haha, well, because `aws_config` is a object with two fields